components/atomic/buttons/ButtonControl/patterns/ButtonControlConfirm.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { forwardRef } from "react";
+import { ButtonControl } from "components/atomic";
+import { useDialogState } from "reakit/Dialog";
+import Modal from "components/layout/Modal";
+import ConfirmModal from "components/composed/ConfirmModal";
+import * as Styled from "../ButtonControl.styles";
+
+type BaseProps = React.ComponentProps<typeof ButtonControl>;
+type ModalProps = React.ComponentProps<typeof Modal>;
+
+/**
+ * A button control that opens a modal
+ */
+const ButtonControlConfirm = forwardRef(
+  (
+    {
+      modalLabel,
+      modalBody,
+      icon,
+      children,
+      onClick,
+      "aria-label": actionLabel,
+      closeDropdown,
+    }: Props,
+    ref
+  ) => {
+    const dialog = useDialogState({ visible: false, animated: true });
+
+    const handleClick = (e: React.MouseEvent) => {
+      e.preventDefault();
+      if (closeDropdown) {
+        closeDropdown();
+      }
+      dialog.show();
+    };
+
+    return (
+      <>
+        <Styled.Disclosure {...dialog} onClick={handleClick}>
+          <ButtonControl as="a" icon={icon} ref={ref}>
```

Comment:
@1aurend This will render an a tag wrapped in a button - since we're using the ButtonControl as just styles (and not an a tag), try `as="span"` and see if it behaves correctly.